src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,692 @@
+        Returns the scale of the image. The standard calculation for the image is based
+        on a size of (1024, 768) so we require these scaling factors.
+        Args:
+            x_size: the x size of the image, in pixels
+            y_size: the y size of the image, in pixels
+        Returns:
+
+        """
+        return _X_SCALING_FACTOR / x_size, _Y_SCALING_FACTOR / y_size
+
+    def find_centre(self, max_run_num=3, rotation_points=6):
+        """
+        Will attempt to find the OAV centre using rotation points.
+        I03 gets the number of rotation points from gda.mx.loop.centring.omega.steps which defaults to 6.
+        If it is unsuccessful in finding the points it will try centering a default maximum of 3 times.
+        """
+
+        # on success iteration is set equal to repeat, otherwise it is iterated
+        # the algorithm will try for a maximum of `repeat` times to find the centre
+        run_num = 0
+        while run_num < max_run_num:
+
+            # do omega spin and harvest edge information
+            (
+                x_positions,
+                y_positions,
+                widths,
+                omega_angles,
+                mid_lines,
+                tip_x_positions,
+                tip_y_positions,
+            ) = tuple(
+                self.get_spacial_data_from_pin_at_different_rotations(rotation_points)
+            )
+
+            (
+                index_of_largest_width,
+                indices_orthogonal_to_largest_width,
+            ) = self.find_widest_point_and_orthogonal_point(
+                x_positions, y_positions, widths, omega_angles
+            )
+
+            x = x_positions[index_of_largest_width]
+            y = y_positions[index_of_largest_width]
+            best_omega_angle = omega_angles[index_of_largest_width]
+
+            # get the angle sufficiently orthogonal to the best omega and
+            # store its mid line - which will be the magnitude in the z axis on 90 degree rotation
+            index_orthogonal_to_largest_width = indices_orthogonal_to_largest_width[-1]
+            best_omega_angle_orthogonal = omega_angles[
+                index_orthogonal_to_largest_width
+            ]
+            z = mid_lines[index_orthogonal_to_largest_width][x]
+
+            # we need to store the tips of the angles orthogonal-ish to the best angle
+            orthogonal_tips_x = tip_x_positions[indices_orthogonal_to_largest_width]
+
+            # best_omega_angle_90 could be zero, which used to cause a failure - d'oh!
+            if best_omega_angle_orthogonal is None:
+                LOGGER.error("Unable to find loop at 2 orthogonal angles")
+                return
+
+            # extract the max_tip_distance again as it could have changed
+            self.oav_parameters.max_tip_distance = (
+                self.oav_parameters._extract_dict_parameter(
+                    "max_tip_distance", reload_json=True
+                )
+            )
```

Comment:
I remember asking when I was thinking about the parameters setup and they said it could indeed change, but it really didn't seem like something that could to me...

I'll ask again now I'm more familiar with it.